labop_convert/autoprotocol/autoprotocol_specialization.py

```diff
@@ -3,25 +3,49 @@
 import uuid
 from typing import Dict
 
+import coordinate_rect_to_row_col_pairs
 import sbol3
 import transcriptic
 import tyto
 from autoprotocol import container_type as ctype
-from autoprotocol.container import WellGroup
+from autoprotocol.container import Container, WellGroup
 from autoprotocol.instruction import Provision, Spectrophotometry
 from autoprotocol.protocol import Protocol
 from autoprotocol.unit import Unit
 from container_api.client_api import matching_containers, strateos_id
 
 import labop
-import labop_convert.autoprotocol.plate_coordinates as pc
+import labop.utils.plate_coordinates as pc
+from labop.strings import Strings
 from labop_convert.autoprotocol.strateos_api import StrateosAPI
 from labop_convert.behavior_specialization import BehaviorSpecialization, ContO
 
 l = logging.getLogger(__file__)
 l.setLevel(logging.ERROR)
 
 
+"""
+Autoprotocol specific extensions for Autoprotocol containers
+"""
+
+
+def coordinate_rect_to_well_group(container: Container, coordinates: str):
+    indices = coordinate_rect_to_row_col_pairs(coordinates)
+    wells = []
+
+    try:
+        for i, j in indices:
+            well = container.well_from_coordinates(i, j)
+            wells.append(well)
+        # wells = [container.well_from_coordinates(i, j) for i, j in indices]
+    except Exception as e:
+        print(container.container_type)
+        print(i, j)
+        print(container.to_dict())
+        raise e
```

Comment:
This try..except clause is a debugging artifact.  That's my fault for leaving that in.

Reply:
removed